### DISCUSSION TITLE
Are dynamic Prisma (client) instances supported?

### DISCUSSION THREAD

**Discussion by tomi-bigpi** [hayes/pothos#380] [Q&A] [answered]
Background:
We're using AWS RDS with IAM authentication which (currently, due to Prisma limitations) requires creating a new Prisma client every 15 minutes or so. We currently have an implementation where the context returns the "current" Prisma instance which is automatically (re)created when the authentication password expires.

Question:
Does Pothos cache the prisma instance given as options to `SchemaBuilder` and/or is there a way to make sure each time Prisma is needed that the latest instance is used to allow dynamically replacing the Prisma instance for this use case?

**Comment by hayes** [comment, marked as answer]
This is not currently supported, but would be easy to add

**Reply by hayes** [reply]
Looked at this a bit.  There isn't an easy way to hack around this manually.  Need to refactor some things to make this work.  Parts of the client get cached in closures during the build step.  The plugin extracts data about relationships from the client during build, and that logic is somewhat tied together right now.  Breaking out the schema info so the client can be loaded dynamically won't be too hard.  I want to make sure that the method for fetching the current client has access to the context object which isn't available during build.  I can probably add this some time in the next week

**Reply by tomi-bigpi** [reply]
Awesome, I'll keep an eye out on future releases.

**Reply by hayes** [reply]
just merged a change that allows you to dynamically get the prisma client on a per-request basis

**Reply by hayes** [reply]
updated docs here: https://pothos-graphql.dev/docs/plugins/prisma#set-up-the-builder

**Reply by tomi-bigpi** [reply]
Do you have a clean way to get the `prisma.dmmf` dynamically in mind?

When my `builder.ts` is imported, the `Prisma` instance is `null` since we need to create it asynchronously by first calling AWS for credentials etc. I'm currently looking at wrapping all `builder.xxx` calls in either functions or classes to control passing in the builder instance, but that's less clean that just doing `import 'types/MyType.ts'`.

**Reply by hayes** [reply]
You probably need to create a client with a fake connection url.   I don't think prisma will create a connection until the client is used, so that should work fine
